src/libraries/System.Net.Http/src/System/Net/Http/Headers/CacheControlHeaderValue.cs

```diff
@@ -27,103 +26,128 @@ public class CacheControlHeaderValue : ICloneable
 
         private static readonly GenericHeaderParser s_nameValueListParser = GenericHeaderParser.MultipleValueNameValueParser;
 
-        private bool _noCache;
+        [Flags]
+        private enum Flags : int
+        {
+            None = 0,
+            MaxAgeHasValue = 1 << 0,
+            SharedMaxAgeHasValue = 1 << 1,
+            MaxStaleLimitHasValue = 1 << 2,
+            MinFreshHasValue = 1 << 3,
+            NoCache = 1 << 4,
+            NoStore = 1 << 5,
+            MaxStale = 1 << 6,
+            NoTransform = 1 << 7,
+            OnlyIfCached = 1 << 8,
+            Public = 1 << 9,
+            Private = 1 << 10,
+            MustRevalidate = 1 << 11,
+            ProxyRevalidate = 1 << 12,
+        }
+
+        private Flags _flags;
         private TokenObjectCollection? _noCacheHeaders;
-        private bool _noStore;
-        private TimeSpan? _maxAge;
-        private TimeSpan? _sharedMaxAge;
-        private bool _maxStale;
-        private TimeSpan? _maxStaleLimit;
-        private TimeSpan? _minFresh;
-        private bool _noTransform;
-        private bool _onlyIfCached;
-        private bool _publicField;
-        private bool _privateField;
+        private TimeSpan _maxAge;
+        private TimeSpan _sharedMaxAge;
+        private TimeSpan _maxStaleLimit;
+        private TimeSpan _minFresh;
         private TokenObjectCollection? _privateHeaders;
-        private bool _mustRevalidate;
-        private bool _proxyRevalidate;
         private UnvalidatedObjectCollection<NameValueHeaderValue>? _extensions;
 
+        private void SetTimeSpan(ref TimeSpan fieldRef, Flags flag, TimeSpan? value)
+        {
+            if (value is null)
+            {
+                fieldRef = default;
+                _flags &= ~flag;
```

Comment:
Any threading concerns?  I assume the answer is "no, it's not safe to manipulate this header type from multiple threads concurrently", but I'm calling it out because this introduces new failure modes if someone does, e.g. what was unrelated data might now be overwritten by a thread writing to other data.

Reply:
> I assume the answer is "no, it's not safe to manipulate this header type from multiple threads concurrently"

Yeah, it feels odd that someone would be modifying different properties on these types concurrently.
I can sprinkle in some interlocked to make it work like it does currently (that is, "it works" as long as you don't access the same property concurrently).